viewing_party/party.py

```diff
@@ -1,23 +1,141 @@
-# ------------- WAVE 1 --------------------
-
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {
+        "title": title, 
+        "genre": genre, 
+        "rating": rating
+    }
+    if not title or not genre or not rating:
+        new_movie = None
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_to_watch):
+    for i in range(len(user_data["watchlist"])):
+        if movie_to_watch in user_data["watchlist"][i]["title"]:
+            del user_data["watchlist"][i]
+            user_data["watched"].append(movie_to_watch)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
 
+    total_rating = 0
+    count = len(user_data["watched"])
+    if user_data["watched"]:
+        for i in range(count):
+            total_rating += user_data["watched"][i]["rating"]
+        return total_rating/count
+    else:
+        return 0

```

Comment:
👍 Nice work! Consider checking for an empty watched list to switch the conditional around and remove the `else` clause. Because the `else` clause contains only one line of code, it seems like a 'dangly' bit of code at the end of all the important logic. We can utilize a guard clause to clean up our code like so:

```python
def get_watched_avg_rating(user_data):
    if not user_data["watched"] or not user_data:
        return 0 
        
    total_rating = 0
    count = len(user_data["watched"])

    for i in range(count):
        total_rating += user_data["watched"][i]["rating"]
    return total_rating/count
```

Also `i` should be a more descriptive name.